### PR TITLE
feat: caching.feature suite, upgraded flagd & tweaks to evaluation.feature suite

### LINKED ISSUE #14: Add new test suite for caching
As caching is being implemented in flagd providers it'd be good to have integration tests for asserting the behaviour is as expected when wired up.

### PR DESCRIPTION
**Pull request by skyerus**
## This PR

- Adds caching.feature suite
- Upgraded flagd to v0.3.0
- evaluation.feature suite fixes based on flagd upgrade

### Related Issues

Fixes #14

### CODE REVIEW

**Review by james-milligan — APPROVED** [review]

**Review by Kavindu-Dodan — APPROVED** [review]

File: features/caching.feature [hunk 1]
```diff
@@ -0,0 +1,71 @@
+Feature: Flag caching
+
+  Background:
+    Given an openfeature client is registered with cache enabled
```
Comment (toddbaert) [inline features/caching.feature:4]: Nice use of "Background".

File: features/evaluation.feature [hunk 2]
```diff
@@ -1,5 +1,8 @@
 Feature: Flag evaluation
 
+  Background:
+    Given an openfeature client is registered with cache disabled
+
```
Comment (toddbaert) [inline features/evaluation.feature:5]: Is this necessary to mention?

File: features/evaluation.feature [hunk 3]
```diff
@@ -1,5 +1,8 @@
 Feature: Flag evaluation
```
Comment (toddbaert) [inline features/evaluation.feature:1]:
```suggestion
Feature: Flag evaluation

# This test suite contains scenarios to test the flag evaluation API.

```

File: features/evaluation.feature (hunk 2 above)
Comment (skyerus) [reply features/evaluation.feature:5]: Do you mean is it necessary to mention at all or is it necessary to include `with cache disabled`?

Comment (toddbaert) [reply features/evaluation.feature:8]: I'm specifically wondering about `with cache disabled`. This is the "vanilla" test suite, I'd like to not pollute it with any concepts like caching.

I guess you're worried that enabling caching here will cause unexpected behavior in the test suite?

Comment (skyerus) [reply features/evaluation.feature:8]: Exactly that, as caching is enabled by default this causes unexpected behaviour, I thought it best to explicitly disable it.

**Pull request merged**